add seloger.com scraper

mazen-r opened the pull request.

Granitosaurus commented on seloger-scraper/seloger.py:
```diff
@@ -0,0 +1,99 @@
+    cards = []
+    # validate the cards data to avoid advertisement cards
+    for card in datasets[0]["cards"]["list"]:
+        if len(card) > 12:
```
so the cards with <= 12 datapoint keys are ads? Maybe there's a better way to identify this? Seems a bit flimsy.

Granitosaurus replied:
so the cards with <= 12 datapoint keys are ads? Maybe there's a better way to identify this? Seems a bit flimsy.

Granitosaurus commented on seloger-scraper/run.py:
```diff
@@ -0,0 +1,39 @@
+    search_data = await seloger.scrape_search(
+        url="https://www.seloger.com/immobilier/achat/immo-bordeaux-33/bien-appartement/",
+        scrape_all_pages=False,
+        max_pages=800,
```
we should limit this to fewer pages

Granitosaurus commented on seloger-scraper/seloger.py:
```diff
@@ -0,0 +1,99 @@
+    # scrape all available pages in the search if scrape_all_pages = True or max_pages > total_search_pages
+    else:
+        total_pages = total_search_pages
+    print("-----------------", total_pages, "----------------")
```
print debug leftover

Granitosaurus commented on seloger-scraper/seloger.py:
```diff
@@ -0,0 +1,99 @@
+    scrape_all_pages: bool,
+    max_pages: int = 10,
+) -> List[Dict]:
+    """scrape seloger search page"""
```
can we add exmaple search URL so people can understand the function faster? Same for `scrape_property`

Granitosaurus reviewed (commented):
lgtm though left some minor comments

mazen-r commented:
Done ✅
Also, the property page suddenly has become 404. I have updated the property page URL and added some nullable fields to allow some missing fields on the new property page.
